diff --git a/src/Utility/LinkitHelper.php b/src/Utility/LinkitHelper.php
index e1ba0f7..222d478 100644
--- a/src/Utility/LinkitHelper.php
+++ b/src/Utility/LinkitHelper.php
@@ -66,6 +66,9 @@ class LinkitHelper {
       return NULL;
     }
 
+    // Remove encoding, like in file names with spaces.
+    $input = urldecode($input);
+
     $host = parse_url($input, PHP_URL_HOST);
     $scheme = parse_url($input, PHP_URL_SCHEME);
 
