diff --git a/src/SmartDateTrait.php b/src/SmartDateTrait.php
index 27d13b4..baf33ee 100644
--- a/src/SmartDateTrait.php
+++ b/src/SmartDateTrait.php
@@ -647,7 +647,7 @@ trait SmartDateTrait {
       }
       $valid_tokens = [];
       // Check for workable day or month tokens.
-      preg_match_all('/(?<!\\\)[djDNlwzSFmMn]/', $settings['date_format'], $valid_tokens, PREG_OFFSET_CAPTURE);
+      preg_match_all('/(?<!\\\)[djDNlwzSFmMnTc]/', $settings['date_format'], $valid_tokens, PREG_OFFSET_CAPTURE);
       if (!$valid_tokens) {
         return $range;
       }
